src/bp/nlu-server/api.ts

```diff
@@ -0,0 +1,237 @@
+  const app = express()
+
+  // This must be first, otherwise the /info endpoint can't be called when token is used
+  app.use(cors())
+
+  app.use(bodyParser.json({ limit: '250kb' }))
+
+  app.use((req, res, next) => {
+    res.header('X-Powered-By', 'Botpress')
+    debugRequest('incoming ' + req.path, { ip: req.ip })
+    next()
+  })
+
+  app.use(monitoringMiddleware)
+  app.use(handleUnexpectedError)
+
+  if (process.core_env.REVERSE_PROXY) {
+    app.set('trust proxy', process.core_env.REVERSE_PROXY)
+  }
+
+  if (options.limit > 0) {
+    app.use(
+      rateLimit({
+        windowMs: ms(options.limitWindow),
+        max: options.limit,
+        message: 'Too many requests, please slow down'
+      })
+    )
+  }
+
+  if (options.authToken?.length) {
+    app.use(authMiddleware(options.authToken))
+  }
+
+  return app
+}
+
+export default async function(options: APIOptions) {
+  const app = createExpressApp(options)
+  const logger = new NLUServerLogger('API')
+  const loggerWrapper: NLU.Logger = {
+    info: (msg: string) => logger.info(msg),
+    warning: (msg: string, err?: Error) => (err ? logger.attachError(err).warn(msg) : logger.warn(msg)),
+    error: (msg: string, err?: Error) => (err ? logger.attachError(err).error(msg) : logger.error(msg))
+  }
+
+  let engine: Engine
+  let modelService: ModelService
+  let trainSessionService: TrainSessionService
+  let trainService: TrainService
+  try {
+    engine = new Engine('nlu-server', loggerWrapper)
+    modelService = new ModelService(options.modelDir)
+    await modelService.init()
+    trainSessionService = new TrainSessionService()
+    trainService = new TrainService(logger, engine, modelService, trainSessionService)
+  } catch (err) {
+    logger.attachError(err).error('an error occured while initializing the server')
+    process.exit(1)
+  }
+
+  app.get('/info', (req, res) => {
+    res.send({
+      version: options.version
+    })
+  })
+
+  const router = express.Router({ mergeParams: true })
+  router.post('/train', async (req, res) => {
+    try {
+      const input: TrainInput = await validate(req.body, TrainInputCreateSchema, {
+        stripUnknown: true
+      })
+
+      const { topics, entities, language, password } = input
+      const intents = _.flatMap(Object.values(topics))
+      const modelHash = engine.computeModelHash(intents, input.entities, input.language)
+
+      const seed = input.seed ?? Math.round(Math.random() * 10000)
+      const modelId = modelService.makeModelId(modelHash, input.language, seed)
+      const modelFileName = modelService.makeFileName(modelId, password)
+
+      // return the modelId as fast as possible
+      // tslint:disable-next-line: no-floating-promises
+      trainService.train(modelFileName, intents, entities, language, seed)
+
+      return res.send({
+        success: true,
+        modelId
+      })
+    } catch (err) {
+      res.status(500).send({
+        success: false,
+        error: err.message
+      })
```

Comment:
Being a bit picky, can you make your return objects in a single line ? I find this to be a lot easier to read.